storage/replica.go

```diff
@@ -1210,6 +1218,11 @@ func (r *Replica) resolveIntents(ctx context.Context, intents []proto.Intent) {
 		// If the intent isn't (completely) local, we'll need to send an external request.
 		// We'll batch them all up and send at the end.
 		if !local {
+			// HACK(tschottdorf): hope that all intents are for the same Txn.
```

Comment:
Spencer is fixing this in his WIP, so I'm not going to bother. Instead will just cherry-pick those parts from him.